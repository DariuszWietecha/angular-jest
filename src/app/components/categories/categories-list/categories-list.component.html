<div class="text-right">
	<a [routerLink]="['/categories/create/']" class="btn btn-primary">Create</a>
</div>

<h2>Categories</h2>
<div class="d-flex flex-wrap">
  <div *ngFor="let category of categories" class="card col-4">
    <div class="card-body">
      <h4 class="card-title">{{ category.name }}</h4>
      <a [routerLink]="['/categories/edit', category.id]"  class="btn btn-primary">Edit</a>
    </div>
  </div>
</div>