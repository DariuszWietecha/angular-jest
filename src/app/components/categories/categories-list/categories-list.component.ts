import { Component, OnInit } from '@angular/core';
import { ICategory } from '../../../services/categories.service';
import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';
import { IStore, selectCategoriesData } from '../../../reducers';
import { loadCompanies } from '../../../actions/companies.actions';
import { loadCategories } from 'src/app/actions/categories.actions';

@Component({
  selector: 'app-categories-list',
  templateUrl: './categories-list.component.html',
  styleUrls: ['./categories-list.component.scss']
})
export class CategoriesListComponent implements OnInit {
  categories$: Observable<ICategory[]> = this.store.select(selectCategoriesData);
  constructor(
    private store: Store<IStore>
  ) { }

  ngOnInit() {
    this.store.dispatch(loadCategories());
  }

}
