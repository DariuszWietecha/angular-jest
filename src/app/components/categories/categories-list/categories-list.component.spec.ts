import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientModule } from '@angular/common/http';
import { RouterTestingModule } from '@angular/router/testing';
import { CategoriesService } from '../../../services/categories.service';
import { CategoriesListComponent } from './categories-list.component';

describe('CategoriesListComponent', () => {
  let component: CategoriesListComponent;
  let fixture: ComponentFixture<CategoriesListComponent>;
  const categoriesListMock = require('../../../../../test/data/categories/categories-get.json');

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientModule, RouterTestingModule],
      declarations: [CategoriesListComponent],
      providers: [CategoriesService],
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CategoriesListComponent);
    component = fixture.componentInstance;
    component.categories = categoriesListMock;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
