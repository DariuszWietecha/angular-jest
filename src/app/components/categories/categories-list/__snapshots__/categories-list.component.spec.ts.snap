// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesListComponent snapShot 1`] = `
<app-categories-list
  categories={[Function Array]}
  categoriesService={[Function CategoriesService]}
>
  <div
    class="text-right"
  >
    <a
      class="btn btn-primary"
      href="/categories/create"
      ng-reflect-router-link="/categories/create/"
    >
      Create
    </a>
  </div><h2>
    Categories
  </h2><div
    class="d-flex flex-wrap"
  >
    
    <div
      class="card col-4"
    >
      <div
        class="card-body"
      >
        <h4
          class="card-title"
        >
          Big Category
        </h4>
        <a
          class="btn btn-primary"
          href="/categories/edit/a0308830-660b-11ea-9266-adf5decb4537"
          ng-reflect-router-link="/categories/edit,a0308830-660b"
        >
          Edit
        </a>
      </div>
    </div>
    <div
      class="card col-4"
    >
      <div
        class="card-body"
      >
        <h4
          class="card-title"
        >
          Small Category
        </h4>
        <a
          class="btn btn-primary"
          href="/categories/edit/eae597d0-660b-11ea-9266-adf5decb4537"
          ng-reflect-router-link="/categories/edit,eae597d0-660b"
        >
          Edit
        </a>
      </div>
    </div>
  </div>
</app-categories-list>
`;
